ndnfs/daemon/driver.go

```diff
@@ -90,22 +270,95 @@ func (d NdnfsDriver) Mount(r volume.MountRequest) volume.Response {
 
 func (d NdnfsDriver) Path(r volume.Request) volume.Response {
 	log.Info(DN, "Path volume: ", r.Name, " Options: ", r.Options)
-	mnt := fmt.Sprintf("%s%s", d.Client.Config.Mountpoint, r.Name)
+	mnt := fmt.Sprintf("%s%s", d.Config.Mountpoint, r.Name)
 	return volume.Response{Mountpoint: mnt}
 }
 
 func (d NdnfsDriver) Remove(r volume.Request) volume.Response {
 	log.Info(DN, "Remove volume: ", r.Name, " Options: ", r.Options)
 	d.Mutex.Lock()
-	d.Client.DeleteVolume(r.Name)
 	defer d.Mutex.Unlock()
+	nfsList, err := d.GetNfsList()
+	if err != nil {
+		log.Panic("Error getting nfs list", err)
+	}
+	var path, service string
+	for i := range(nfsList) {
+		if strings.Contains(nfsList[i], r.Name) {
+			path = nfsList[i]
+		}
+	}
+	if os.Getenv("CCOW_SVCNAME") != "" {
+		service = os.Getenv("CCOW_SVCNAME")
+	} else {
+		service = d.Config.Servicename
+	}
+	data := make(map[string]interface{})
+	data["serve"] = path
+	url := fmt.Sprintf("service/%s/serve", service)
+	_, err = d.Request("DELETE", url, data)
+	if err != nil {
+		log.Panic("Error while handling request", err)
+	}
+
+	parts := strings.Split(path, "/")
+	url = fmt.Sprintf("clusters/%s/tenants/%s/buckets/%s", parts[0], parts[1], parts[2])
+	_, err = d.Request("DELETE", url, nil)
```

Comment:
A few lines about err is fatal. In this case 'err' is just thrown away. At line 309 it is a new variable with the same name is declared